src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -117,7 +117,7 @@ internal partial class Binder
                     hasErrors = true;
                 }
 
-                source = ConvertSwitchExpression((BoundUnconvertedSwitchExpression)source, type, diagnostics, hasErrors);
+                source = ConvertSwitchExpression((BoundUnconvertedSwitchExpression)source, type, targetTyped: false, diagnostics, hasErrors);
```

Comment:
No, because it isn't a switch expression conversion.  It is some other conversion that uses the switch expression's natural type as input.

---
In reply to: [318803559](https://github.com/dotnet/roslyn/pull/38187#discussion_r318803559) [](ancestors = 318803559)